x-pack/plugin/esql/src/internalClusterTest/java/org/elasticsearch/xpack/esql/action/EsqlActionIT.java

```diff
@@ -1220,6 +1219,34 @@ public void testUnsupportedTypesOrdinalGrouping() {
         }
     }
 
+    public void testStatsNestFields() {
+        String node1 = internalCluster().startDataOnlyNode();
+        String node2 = internalCluster().startDataOnlyNode();
+        assertAcked(
+            client().admin()
+                .indices()
+                .prepareCreate("index-1")
+                .setSettings(Settings.builder().put("index.routing.allocation.require._name", node1))
+                .setMapping("field_1", "type=integer")
+        );
+        assertAcked(
+            client().admin()
+                .indices()
+                .prepareCreate("index-2")
+                .setSettings(Settings.builder().put("index.routing.allocation.require._name", node2))
+                .setMapping("field_2", "type=integer")
+        );
+        try (
+            var resp = run("from index-1,index-2 | where field_1 is not null | stats c = count(*), c1 = count(field_1), m = min(field_1)")
```

Comment:
This test fails because the final min aggregation expects the local aggregation to return an int. Instead it returns null since the field is missing.
This causes the following failure:
```
  2> REPRODUCE WITH: gradlew ':x-pack:plugin:esql:internalClusterTest' --tests "org.elasticsearch.xpack.esql.action.EsqlActionIT.testStatsNestFields" -Dtests.seed=B786427DD46396B2 -Dtests.locale=sk-SK -Dtests.timezone=America/Argentina/ComodRivadavia -Druntime.java=17
  2> java.lang.NullPointerException: Cannot invoke "org.elasticsearch.compute.data.IntVector.getPositionCount()" because "min" is null
        at __randomizedtesting.SeedInfo.seed([B786427DD46396B2:E62DFB5E322E8A9D]:0)
        at org.elasticsearch.compute.aggregation.MinIntAggregatorFunction.addIntermediateInput(MinIntAggregatorFunction.java:99)
        at org.elasticsearch.compute.aggregation.Aggregator.processPage(Aggregator.java:40)
        at org.elasticsearch.compute.operator.AggregationOperator.addInput(AggregationOperator.java:79)
        at org.elasticsearch.compute.operator.Driver.runSingleLoopIteration(Driver.java:191)
        at org.elasticsearch.compute.operator.Driver.run(Driver.java:129)
        at org.elasticsearch.compute.operator.Driver$1.doRun(Driver.java:285)
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
        at org.elasticsearch.common.util.concurrent.TimedRunnable.doRun(TimedRunnable.java:33)
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:983)
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:833)
```

@nik9000 am I missing something? Looks to me that intermediate nulls are not handled inside ungrouped aggs but they are inside grouped ones.

Reply:
Raised https://github.com/elastic/elasticsearch/issues/100928